test/ansatz_generation/test_parametrize_circuit.py

```diff
@@ -0,0 +1,35 @@
+# This code is a Qiskit project.
+#
+# (C) Copyright IBM 2025.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+import pytest
+from qiskit.circuit.random import random_circuit
+from qiskit.quantum_info import Operator, Statevector, process_fidelity, state_fidelity
+
+from qiskit_addon_aqc_tensor import parametrize_circuit
+
+# pylint: disable=no-self-use
+
+
+class TestParametrizeCircuit:
```

Comment:
Would be nice to add a test of a circuit that already has _some_ unbound parameters, perhaps even ones that are reference more than once in the circuit.

Reply:
Done. As we discussed separately, the returned `initial_params` contains `None` for parameters that were already in the circuit.